backend/event/views.py

```diff
@@ -0,0 +1,298 @@
+from django.http import JsonResponse
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework import viewsets, filters, status
+from rest_framework.permissions import IsAuthenticated, BasePermission
+from django.db.models import Q
+from django.utils import timezone
+from .models import Event, Invitee, Notification
+from .serializers import EventSerializer, InviteeSerializer, NotificationSerializer
+from django.utils.timezone import now
+from .models import InviteeEventView
+from .serializers import InviteeEventViewSerializer
+from django.db import transaction
+from django.contrib.auth.models import User
+from django.contrib.auth import get_user_model
+from django.conf import settings
+from rest_framework.views import APIView
+
+User=get_user_model()
+
+class IsEventOwner(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        return obj.created_by == request.user
+
+class EventViewSet(viewsets.ModelViewSet):
+    queryset = Event.objects.all().order_by('-created_at')
+    serializer_class = EventSerializer
+    filter_backends = [filters.SearchFilter, filters.OrderingFilter]
+    search_fields = ['title', 'location']
+    permission_classes = [IsAuthenticated, IsEventOwner]
+
+    def perform_create(self, serializer):
+
+        event = serializer.save(created_by=self.request.user)
+
+
+        Invitee.objects.create(
```

Comment:
@sanari2019 I prefer it when the Serializer handles the additional creations it keeps the views simple.